rocketpy/simulation/flight.py

```diff
@@ -1120,6 +1134,23 @@ def __init_equations_of_motion(self):
             # NOTE: The u_dot is faster, but only works for solid propulsion
             self.u_dot_generalized = self.u_dot
 
+    def __init_controllers(self):
+        """Initialize controllers"""
+        self._controllers = self.rocket._controllers[:]
+        if self._controllers:
+            if self.time_overshoot == True:
+                self.time_overshoot = False
+                warnings.warn(
+                    "time_overshoot has been set to False due to the presence of controllers. "
+                )
+            # reset controllable object to initial state (only airbrakes for now)
+            for air_brakes in self.rocket.air_brakes:
+                air_brakes._reset()
```

Comment:
Right now I am not completely against this implementation, since this is a bugfix. However, this pose for us some scalability issues that we have tried to avoid while implementing the controllers.

This code block requires that the `Flight` knows everything that is controllable in a "hardcoded" manner.

Ideally, we would request to the controller of those objects to `reset` them. This resonates quite well with "real life" IMHO.

@MateusStano just to raise a bit of discussion, would it work implementing a `reset` method in the `Controller` class that calls the `controller_function` with `initial_observed_variables` to reset everything this controller controls?

I understand this might have unintended repercutions, since the `controller_function` is user implemented, but I would like your opinion on this because it would bring great architectural benefits in the future.

Reply:
> This code block requires that the `Flight` knows everything that is controllable in a "hardcoded" manner.

This particular point is not an issue I think. When adding controllable objects to the Rocket class we would need to keep track of it there. This `for air_brakes in self.rocket.air_brakes` would just become `for c in self.rocket.controllables` or something similar.

> , would it work implementing a `reset` method in the `Controller` class that calls the `controller_function` with `initial_observed_variables` to reset everything this controller controls?

No it wouldn't. We can't call the controller function because the user could have dynamic variables that change at every call, and having an extra call not related to the simulation would be confusing I think


> Ideally, we would request to the controller of those objects to `reset` them. This resonates quite well with "real life" IMHO.

This makes more sense though. I think maybe the `._reset` method should actually be in the controller class, and the controller should receive the initial values and what params it has to reset. I think that should be part of the changes coming to the controller class, to support controlling everything besides just air brakes. That change would require deleting the `AirBrakes._reset` method though. Do you think it is okay leaving the `_reset` method in just for this bug fix? It would be a lot of work to add the other way right now

Reply:
Let's keep the way it is right now. In future PR we implement the `reset` on the controller.